model/cloud/openrouter.go

```diff
@@ -0,0 +1,51 @@
+package cloud
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/sashabaranov/go-openai"
+)
+
+type openRouterProvider struct {
+	// client is the client to query the LLM model.
+	client *openai.Client
+}
+
+// OpenRouterToken is the "OPENROUTER_TOKEN" environment variable.
+var OpenRouterToken = os.Getenv("OPENROUTER_TOKEN")
+
+// NewOpenRouterProvider returns an "openrouter.ai" provider.
+func NewOpenRouterProvider(token string) Provider {
+	config := openai.DefaultConfig(token)
+	config.BaseURL = "https://openrouter.ai/api/v1"
+	client := openai.NewClientWithConfig(config)
+
+	return &openRouterProvider{
+		client: client,
+	}
+}
+
+// Query queries the LLM.
+func (o *openRouterProvider) Query(request string, model string) (response string, err error) {
+	apiResponse, err := o.client.CreateChatCompletion(
+		context.Background(),
```

Comment:
Please make the "context" a parameter (first parameter) so we can do the timeouts from outside right away.